File: app/csharp/AccountingSystem.Api/Domain/Models/Financial/BalanceSheetItem.cs
namespace AccountingSystem.Domain.Models.Financial;

/// <summary>
/// 貸借対照表の項目
/// </summary>
public record BalanceSheetItem
{
    /// <summary>
    /// 勘定科目コード
    /// </summary>
    public required string AccountCode { get; init; }

    /// <summary>
    /// 勘定科目名
    /// </summary>
    public required string AccountName { get; init; }

    /// <summary>
    /// 残高
    /// </summary>
    public required decimal Balance { get; init; }

    /// <summary>
    /// 構成比率（%）
    /// </summary>
    public required decimal Percentage { get; init; }
}

File: app/ruby/accounting_system/app/models/balance_sheet_item.rb
# frozen_string_literal: true

# 貸借対照表の項目（勘定科目ごとの残高と構成比率）
class BalanceSheetItem
  attr_accessor :account_code, :account_name, :balance, :percentage

  def initialize(account_code:, account_name:, balance:, percentage: nil)
    @account_code = account_code
    @account_name = account_name
    @balance = balance
    @percentage = percentage
  end
end
